src/map/mercenary.c

```diff
@@ -260,37 +260,34 @@ static int merc_contract_end_timer(int tid, int64 tick, int id, intptr_t data)
 	}
 
 	md->contract_timer = INVALID_TIMER;
-	mercenary->delete(md, 0); // Mercenary soldier's duty hour is over.
+	mercenary->delete(md, MDT_EXPIRED); // Mercenary soldier's duty hour is over.
 
 	return 0;
 }
 
-static int merc_delete(struct mercenary_data *md, int reply)
+static int merc_delete(struct mercenary_data *md, int type)
```

Comment:
put some comment and describe what this function does
/*
*
*/
something, you can see them on other functions

Reply:
by the way, MERC_DELETE_RUNAWAY is ... trigger when the player dead or hit by SC__GROOMY

```
/**
 * Mercenary Removal Function
 * @param md Mercenary data belongs to the player
 * @param type :
 * - MERC_DELETE_EXPIRED - fulfill mercenary soldier's duty in full duration. Loyalty +1
 * - MERC_DELETE_KILLED - let the mercenary die. Loyalty -1
 * - MERC_DELETE_REMOVED - trigger when player right-click on mercenary, and select remove it
 * - MERC_DELETE_RANAWAY - trigger when player died, or hit by status SC__GROOMY
 * return value is meaningless
 **/

```
---------------------------
```
	if( !sd )
		return unit->free(&md->bl, CLR_OUTSIGHT);
```
this line execute when player logs out, always return 0 from unit_free function